pkg/istioversions/versions.go

```diff
@@ -0,0 +1,142 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istioversions
```

Comment:
nit: can we rename this to use the singular form `istioversion`? Then we'd have the following:
- `istioversion.Default`
- `istioversion.List`
- `istioversion.Map`
- `istioversion.Resolve` (we can rename `ResolveVersion` to `Resolve`)
- ...